src/ssh-connection-manager.ts

```diff
@@ -68,29 +73,44 @@ export class SshConnectionManager {
   }
 
   async getSftp(address: string): Promise<Sftp | undefined> {
+    const username = getUsername(this.configManager, address);
     try {
-      const conn = await this.getConnection(address);
+      const conn = await this.getConnection(address, username);
       return new Sftp(conn);
     } catch (err) {
       if (this.isAuthenticationError(err)) {
-        const username = await this.promptForUsername(address);
-
-        if (username) {
+        this.displayAuthenticationError(err.level, username, address);
+        if (await this.promptForUsername(address)) {
           return await this.getSftp(address);
         }
-
         this.showUsernameRequiredError();
       }
       throw err;
     }
   }
 
+  async displayAuthenticationError(level: string, username: string, address: string) {
+    if (level === 'wrong-user') {
+      vscode.window.showWarningMessage(
+        `The username '${username}' is not valid on host ${address}`
+      );
+    } else {
+      const msg = `We couldn't connect to the node. Please check that the machine is online and your tailnet ACLs allow you to access it"`;
+      const action = await vscode.window.showWarningMessage(msg, 'Learn more');
+      if (action) {
+        vscode.env.openExternal(
+          vscode.Uri.parse('https://tailscale.com/kb/1265/vscode-extension/#troubleshooting')
```

Comment:
Until the VS Code docs are improved, I think this link is better:

```suggestion
          vscode.Uri.parse('https://tailscale.com/kb/1193/tailscale-ssh/#ensure-tailscale-ssh-is-permitted-in-acls')
```